### PR TITLE
BSAPI-7: Creating a 5 star scale rating for a book with a timestamp

### PR DESCRIPTION
**Pull request by airiz04**

### CODE REVIEW

**Comment by airiz04** [comment]
I believe the issue lies in the customer entity class, with two of the variables being mapped to “customer”

File: src/main/java/com/bookstore/bookstoreapi/data/repository/WishlistRepository.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.bookstore.bookstoreapi.data.repository;
 
 import com.bookstore.bookstoreapi.data.entity.Customer;
-import com.bookstore.bookstoreapi.data.entity.Wishlist;
+//import com.bookstore.bookstoreapi.data.entity.Wishlist;
```
Comment (IvanoskiHarmonia) [inline src/main/java/com/bookstore/bookstoreapi/data/repository/WishlistRepository.java]: Please uncomment this, or just run 
```
git checkout HEAD src/main/java/com/bookstore/bookstoreapi/data/repository/WishlistRepository.java
```
 to return this repo interface to its previous state, this is not a solution.

**Comment by IvanoskiHarmonia** [comment]
> I believe the issue lies in the customer entity class, with two of the variables being mapped to “customer”

Yeah this should not be a problem but the fix was 2 things:
1. `@ManyToOne` needs to be annotated with `@JsonIgnore`
2. And also the saving of book1 3 times did not work for one reason or another.

File: src/main/java/com/bookstore/bookstoreapi/data/controller/RatingController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class RatingController {
+
+  @Autowired private RatingService ratingService;
+
+  @PostMapping("/ratings/books")
```
Comment (IvanoskiHarmonia) [inline src/main/java/com/bookstore/bookstoreapi/data/controller/RatingController.java:18]: I am not sure if you want to have this as `ratings/books`, or `rating/book`, the plural suggests that you are rating multiple.

File: src/main/java/com/bookstore/bookstoreapi/data/entity/Rating.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  private Long id;
+
+  @Column(name = "Timestamp")
+  @Builder.Default
+  private String timeStamp =
+      new SimpleDateFormat("MM/dd/yyyy HH:mm:ss").format(new java.util.Date());
+
+  @Column(name = "Rating")
+  @Min(1)
+  @Max(5)
```
Comment (IvanoskiHarmonia) [inline src/main/java/com/bookstore/bookstoreapi/data/entity/Rating.java:43]: have you tested if this actually prevents from making ratings bigger than 5 and smaller than 1?

Comment (airiz04) [reply src/main/java/com/bookstore/bookstoreapi/data/entity/Rating.java:43]: Tested it out, program gives an error when trying smaller than 1 and bigger than 5. 👍

File: src/main/java/com/bookstore/bookstoreapi/data/controller/RatingController.java (hunk 2 above)
Comment (airiz04) [reply src/main/java/com/bookstore/bookstoreapi/data/controller/RatingController.java:18]: I'll keep it as books, makes sense considering many books can be rated.

**Review by IvanoskiHarmonia — APPROVED** [review]
Beautiful work!

**Review by rheaden10 — APPROVED** [review]

File: src/main/java/com/bookstore/bookstoreapi/data/controller/RatingController.java (hunk 2 above)
Comment (IvanoskiHarmonia) [reply src/main/java/com/bookstore/bookstoreapi/data/controller/RatingController.java:18]: I wouldn't say so, can this API call rate multiple books at once?

**Pull request merged**